tests/shared/test_session.py

```diff
@@ -110,17 +107,87 @@ async def make_request(client_session):
             with anyio.fail_after(1):  # Timeout after 1 second
                 await ev_tool_called.wait()
 
-            # Send cancellation notification
-            assert request_id is not None
-            await client_session.send_notification(
-                ClientNotification(
-                    CancelledNotification(
-                        method="notifications/cancelled",
-                        params=CancelledNotificationParams(requestId=request_id),
-                    )
-                )
-            )
+            # Cancel the task via task group
+            tg.cancel_scope.cancel()
 
             # Give cancellation time to process
             with anyio.fail_after(1):
                 await ev_cancelled.wait()
+
+            # Check server cancel notification received
+            with anyio.fail_after(1):
+                await ev_cancel_notified.wait()
+
+            # Give cancellation time to process on server
+            with anyio.fail_after(1):
+                await ev_tool_cancelled.wait()
+
+
+@pytest.mark.anyio
+async def test_request_cancellation_uncancellable():
+    """Test that asserts a call with cancellable=False is not cancelled on
+    server when cancel scope on client is set."""
+
+    ev_tool_called = anyio.Event()
+    ev_tool_commplete = anyio.Event()
+    ev_cancelled = anyio.Event()
+
+    # Start the request in a separate task so we can cancel it
+    def make_server() -> Server:
+        server = Server(name="TestSessionServer")
+
+        # Register the tool handler
+        @server.call_tool()
+        async def handle_call_tool(name: str, arguments: dict | None) -> list:
+            nonlocal ev_tool_called, ev_tool_commplete
+            if name == "slow_tool":
+                ev_tool_called.set()
+                with anyio.CancelScope():
+                    with anyio.fail_after(10):  # Long enough to ensure we can cancel
+                        await ev_cancelled.wait()
+                    ev_tool_commplete.set()
+                    return []
+
+            raise ValueError(f"Unknown tool: {name}")
+
+        # Register the tool so it shows up in list_tools
+        @server.list_tools()
+        async def handle_list_tools() -> list[types.Tool]:
+            return [
+                types.Tool(
+                    name="slow_tool",
+                    description="A slow tool that takes 10 seconds to complete",
+                    inputSchema={},
+                )
+            ]
+
+        return server
+
+    async def make_request(client_session: ClientSession):
+        nonlocal ev_cancelled
+        try:
+            await client_session.call_tool(
+                "slow_tool",
+                cancellable=False,
+                read_timeout_seconds=timedelta(seconds=10),
```

Comment:
A test that validates a short timeout with an uncancelable call would be sensible